### PR TITLE
DYN-7433 Cleanup Node Layout Crash

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Crash when we have a pin inside a group and using the GraphLayout functionality.
When trying to add an Edge was using nodes that doesn't exist in the GraphLayout so it was crashing when trying to access the node coordinates (there is no null validations) so I've added a validation that check if the two nodes exist before trying to create the Edge. I noticed that there is a re-organization layout problem when using the CleanNodeLayout functionality and having a pin inside a group or in the Canvas (seems that the wires and pins are re-organized in a weird way) but I consider that we should create a task for this specific case (just if is a bug, due that pins behave weird when re-organizing).

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Crash when we have a pin inside a group and using the GraphLayout functionality.

### Reviewers

@QilongTang 

### FYIs

@achintyabhat

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
This is a GIF showing the behavior BEFORE my fix (crashing)
DynamoSandbox_6a0Tg7kRCr

**Comment by RobertGlobant20** [comment]
This is the behavior after my fix (no crash)
OEBIncK4RQ

File: src/Engine/GraphLayout/GraphLayout.cs [hunk 1]
```diff
@@ -108,6 +108,9 @@ public void AddNode(Guid guid, double width, double height, double x, double y,
         /// <param name="endY">The y coordinate of the connector's right end point.</param>
         public void AddEdge(Guid startId, Guid endId, double startX, double startY, double endX, double endY)
         {
+            //Validates that the two nodes that will be used to create the Edge exist
+            if (Nodes.Where(node => node.Id == startId).Count() == 0  || Nodes.Where(node => node.Id == endId).Count() == 0) return;
```
Comment (QilongTang) [inline src/Engine/GraphLayout/GraphLayout.cs:112]: Maybe use !Any() instead?

Comment (RobertGlobant20) [reply src/Engine/GraphLayout/GraphLayout.cs]: updated in the next commit: https://github.com/DynamoDS/Dynamo/pull/15495/commits/0f97fe1be9df374243be1cbfe38238364f7bce45

**Pull request merged**